centos_ci.sh

```diff
@@ -353,14 +335,26 @@ else
   export GITHUB_ACCESS_TOKEN=$GITHUB_TOKEN
 
   prepare_repo;
+  setup_kvm_docker_machine_driver;
+  # Navigate to the repo
+  cd $GOPATH/src/github.com/minishift/minishift
 
   if [[ "$JOB_NAME" = "minishift-docs" ]]; then
-    perform_docs_publish;
+    # REPO and BRANCH variables are populated via https://ci.centos.org/job/minishift-docs
+    REPO_OWNER=$(echo $REPO | cut -d"/" -f4)
+    perform_docs_publish "$REPO_OWNER" "minishift/docs/ondemand/$REPO_OWNER-$BRANCH";
   elif [[ "$JOB_NAME" = "minishift-release" ]]; then
-    perform_release $RSYNC_PASSWORD;
+    perform_release "$RSYNC_PASSWORD";
   elif [[ "$JOB_NAME" = "minishift-nightly" ]]; then
     perform_nightly;
-  else
-    build_and_test $RSYNC_PASSWORD;
+  elif [[ "$JOB_NAME" = "minishift" ]]; then
+    make prerelease synopsis_docs link_check_docs
```

Comment:
why don't you put these into functions like perform_master_build and perform_pr_build. That would align better with the rest of the flow.